File: clients/python/coyote/apis/msgs_topic.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    MsgTopicConfigureIn,
    MsgTopicConfigureOut,
)

from ..models.msg_topic_configure_in import _MsgTopicConfigureIn


class MsgsTopicAsync(ApiBase):
    async def configure(
        self,
        topic: str,
        partitions: int,
        msg_topic_configure_in: MsgTopicConfigureIn,
    ) -> MsgTopicConfigureOut:
        """Configures the number of partitions for a topic.

        Partition count can only be increased, never decreased. The default for a new topic is 1."""
        body = _MsgTopicConfigureIn(
            topic=topic,
            partitions=partitions,
        ).model_dump(exclude_unset=True, by_alias=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/msgs/topic/configure",
            body=body,
            response_type=MsgTopicConfigureOut,
        )


class MsgsTopic(ApiBase):
    def configure(
        self,
        topic: str,
        partitions: int,
        msg_topic_configure_in: MsgTopicConfigureIn,
    ) -> MsgTopicConfigureOut:
        """Configures the number of partitions for a topic.

        Partition count can only be increased, never decreased. The default for a new topic is 1."""
        body = _MsgTopicConfigureIn(
            topic=topic,
            partitions=partitions,
        ).model_dump(exclude_unset=True, by_alias=True)

        return self._request_sync(
            method="post",
            path="/api/v1/msgs/topic/configure",
            body=body,
            response_type=MsgTopicConfigureOut,
        )

File: z-clients/java/src/main/java/com/svix/diom/apis/MsgsTopic.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.DiomException;
import com.svix.diom.HttpClient;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;
import com.svix.diom.models.MsgTopicConfigureIn;
import com.svix.diom.models.MsgTopicConfigureOut;
import com.svix.diom.models.MsgTopicConfigureIn_;

public class MsgsTopic {
    private final HttpClient client;

    public MsgsTopic(HttpClient client) {
        this.client = client;
    }

    /**
* Configures the number of partitions for a topic.
* 
* Partition count can only be increased, never decreased. The default for a new topic is 1.
*/
    public MsgTopicConfigureOut configure(
        String topic,
        final MsgTopicConfigureIn msgTopicConfigureIn
    ) throws DiomException {
        MsgTopicConfigureIn_ body = new MsgTopicConfigureIn_(
            msgTopicConfigureIn.getNamespace(),
            topic,
            msgTopicConfigureIn.getPartitions()
        );

        return this.client.executeRequest(
            "POST",
            "/api/v1.msgs.topic.configure",
            null,
            body,
            MsgTopicConfigureOut.class
        );
    }
}